docs/architecture-decisions/2023-06-06_frontend-templates.md

```diff
@@ -13,23 +13,54 @@ AlgoKit v2 aims provide an end-to-end development and deployment experience that
+2. Implement a bundling process for backend and frontend templates in the starter repositories that code duplication and maintenance overhead.
+3. Develop deployment pipelines for frontend components inside end-to-end starter repository that supports TestNet, and MainNet deployments to hosting providers of choice (Netlify, Vercel and etc to be decided).
+4. Ensure that the end-to-end starter repository supports both manual deployment and deployment via a continuous deployment pipeline.
 
 ## Principles
 
 - [AlgoKit Guiding Principles](../../docs/algokit.md#Guiding-Principles) - all of the aforementioned requirements should be met in a way that is consistent with the guiding principles of AlgoKit or attempt to find a balanced trade of between the principles that satisfies the requirements.
 
-## Options
+## Explored options
+
+Enhancing AlgoKit's templating capabilities involves a notable shift from managing individual templates to co-locating multiple templates or projects together. This is a marked increase in complexity that requires careful consideration of the varying trade-offs between different approaches. The subsequent sections explore these options and their associated trade-offs, providing a concise overview of this new challenge and our proposed solutions.
+
+### Option 1: Monorepos
+
+This option suggests that the frontend and backend templates should be bundled together in a single repository. This approach aims to simplify the maintenance of the templates by reducing the number of repositories that need to be maintained. Both official and community templates should be build on top of this approach where frontend is always tailored for specific needs of the smart contracts that are being used.
 
-Expanding AlgoKit templating capabilities is a non-trivial task that requires careful consideration of the trade-offs between the different approaches. The following section outlines the different options that were considered and the trade-offs associated with each of them.
+Drawbacks:
 
-> Refer to git history to see old revisions of the document. After further discussions with the team we are now aligned on the separate repositories approach. Hence the sections below are updated to reflect the new approach.
+- **Lack of flexibility**: The monorepo approach does not allow developers to choose their preferred frontend technology stack. This is because the frontend is tightly coupled with the backend and the two cannot be separated. This is a major drawback as it limits the flexibility of the templates and does not allow developers to choose their preferred frontend technology stack.
+- **Increased maintenance overhead**: The monorepo approach increases the maintenance overhead as it requires the templates to be updated together.
```

Comment:
I'm not sure if I agree with this, I'd say it's actually less maintenance to have one repository? The flexibility is a big issue as is the fact that as soon as you want to have a different (say) frontend variant you would then need to copy all of the smart contract stuff into that second template (that's a maintenance overhead though).

Reply:
Agree with your point, should be refined now to reflect the note